controllers/auth.controller.js

```diff
@@ -1,389 +1,387 @@
+    return next(new AppError(error.message, 400))
+  }
+
+  // check if user exists
+  const existingAdmin = await Admin.findOne({
+    email: req.body.email
+  })
+
+  if (existingAdmin) {
+    return next(
+      new AppError('Admin already exist, please use the login route.', 400)
+    )
+  }
+
+  // hashPassword
+  const hashedPassword = await hashPassword(req.body.password, 10)
+
+  try {
+    // create a new user
     const newAdmin = new Admin({
-         firstName: req.body.firstName,
-         lastName: req.body.lastName,
-         email: req.body.email,
-         password: hashedPassword,
-         photo: {
-            storagePath: `/public/uploads/adminPhoto/${req.file.filename}`,
-            contentType: req.file.mimetype,
-        }
-     });
-      
-      
-     const accessToken = jwt.sign({ 
-         userId: newAdmin._id, userRole: newAdmin.role}, 
-         process.env.JWT_SECRET, {
-         expiresIn: "1h"
-     });
-      
-     newAdmin.accessToken = accessToken;
-     await newAdmin.save();
-     console.log("here3")
-     return res.status(201).json({
-         status: 'success',
-         message: 'Registration successful',
-         newAdmin,
-     });
-    } catch (error) {
-      res.status(500).json({
-         message: "Internal server error",
-         error: error
-      })
-    }
- });
+      firstName: req.body.firstName,
+      lastName: req.body.lastName,
+      email: req.body.email,
+      password: hashedPassword,
+      photo: {
+        storagePath: req.file.path,
+        contentType: req.file.mimetype
+      }
+    })
+
+    const accessToken = jwt.sign(
+      {
+        userId: newAdmin._id,
+        userRole: newAdmin.role
+      },
+      process.env.JWT_SECRET,
+      {
+        expiresIn: '1h'
+      }
+    )
+
+    newAdmin.accessToken = accessToken
+    await newAdmin.save()
+    console.log('here3')
+    return res.status(201).json({
+      status: 'success',
+      message: 'Registration successful',
+      newAdmin
+    })
+  } catch (error) {
+
+    res.status(500).json({
+      message: 'Internal server error',
+      error: error
+    })
+  }
+})
 
 // signin function
-exports.signIn = asyncWrapper(async(req, res) => {
-    // validate user body request
-    const { error } = validateLogin(req.body);
-    if (error) return next(new AppError(error.message, 400));
-    const { email, password } = req.body;
-
-    const user = await User.findOne({ email })
-    if (!user) {
-        return res.status(404).json({
-            message: "User does not exist"
-        })
-    }
-     
-    
-    const correctPassword = await comparePassword(password, user.password)
-    if(!correctPassword) {
-        return res.status(400).json({
-            message: "Incorrect email or password"
-            })
+exports.signIn = asyncWrapper(async (req, res) => {
+  // validate user body request
+  const { error } = validateLogin(req.body)
+  if (error) return next(new AppError(error.message, 400))
```

Comment:
please leave my indentation for me nah...lol

Reply:
it just happens, i use prettier, i think thats why